db/log_reader.cc

```diff
@@ -364,6 +366,7 @@ void Reader::MaybeVerifyPredecessorWALInfo(
       stop_replay_for_corruption_) {
     return;
   }
+  assert(corrupted_wal_number_);
```

Comment:
nit: maybe add assertion for `*corrupted_wal_number_ == kMaxSequenceNumber` during construction.

Reply:
fixed